docker.go

```diff
@@ -100,6 +101,15 @@ func (p Plugin) Exec() error {
 		fmt.Println("Registry credentials not provided. Guest mode enabled.")
 	}
 
+	// pre-pull cache image
+	for _, img := range p.Build.CacheFrom {
+		cmd := commandPull(img)
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		trace(cmd)
+		cmd.Run()
```

Comment:
I think these commands should be appended to the list, after docker info

```diff
cmds = append(cmds, commandInfo())    // docker info

+for _, img := range p.Build.CacheFrom {
+  cmds = append(cmds, commandPull(img)) 
+}
```

Reply:
The problem with that is that it will throw an error when the image/tag doesn't exist yet.

Reply:
Maybe we can append them with `|| true` to avoid build failures? I do that manually in my CI for this specific reason.

Reply:
@Jean85 I appreciate the suggest, however, `||` would not work because that is a shell notation and these commands are not executed with bash. They are executed from Go with the `exec` syscall.

Reply:
@bradrydzewski do you still want me to move the commands?

Reply:
@bradrydzewski ping

Reply:
I think you should move that command yes, the user have to enable the feature. If the name is wrong, the build should fail and the user should correct it

Reply:
It's not a naming issue, its when you build the first time the image does not exist yet.